src/main/java/uk/gov/hmcts/reform/sendletter/services/LetterService.java

```diff
@@ -78,9 +79,13 @@ public UUID send(Letter letter, String serviceName) throws JsonProcessingExcepti
         DbLetter dbLetter = new DbLetter(id, serviceName, letter);
 
         //Save message details to db for reporting
-        letterRepository.save(dbLetter, Instant.now(), messageId);
-        placeLetterInQueue(dbLetter, messageId);
-        return id;
+        SendLetterResponse response = letterRepository.save(dbLetter, Instant.now(), messageId);
+
+        if (response.isUpdated) {
```

Comment:
Should it be the opposite, i.e. if not updated than place in the queue?

Reply:
No, updated flag is true if insert/update happened. Otherwise existing ID is returned on duplicate message id violation - not an update

Reply:
Oh, fair enough.